services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/DigitalGreenCertificateToProtobufMapping.java

```diff
@@ -0,0 +1,108 @@
+  VaccineJsonStringObject readMahJson() {
+    String path = distributionServiceConfig.getDigitalGreenCertificate().getMahJsonPath();
+    return readConfiguredJsonOrDefault(path, "dgc/vaccine-mah.json",
+        VaccineJsonStringObject.class);
+  }
+
+  /**
+   * Read the JSON for the medicinal products.
+   *
+   * @return The corresponding JSON object.
+   */
+  VaccineJsonStringObject readMedicinalProductJson() {
+    String path = distributionServiceConfig.getDigitalGreenCertificate().getMedicinalProductsJsonPath();
+    return readConfiguredJsonOrDefault(path, "dgc/vaccine-medicinal-product.json",
+        VaccineJsonStringObject.class);
+  }
+
+  /**
+   * Read the JSON for the prophylaxis.
+   *
+   * @return The corresponding JSON object.
+   */
+  VaccineJsonStringObject readProphylaxisJson() {
+    String path = distributionServiceConfig.getDigitalGreenCertificate().getProphylaxisJsonPath();
+    return readConfiguredJsonOrDefault(path, "dgc/vaccine-prophylaxis.json",
+        VaccineJsonStringObject.class);
+  }
+
+  /**
+   * Create the Protobuf from JSON.
+   *
+   * @return the protobuf filled with values from JSON.
+   */
+  public ValueSets constructProtobufMapping() {
+    List<ValueSetItem> mahItems = toValueSetItems(readMahJson().getValueSetValues());
+    List<ValueSetItem> productItems = toValueSetItems(readMedicinalProductJson().getValueSetValues());
+    List<ValueSetItem> prophylaxisItems = toValueSetItems(readProphylaxisJson().getValueSetValues());
+
+    return ValueSets.newBuilder()
+        .setMa(ValueSet.newBuilder().addAllItems(mahItems).build())
+        .setVp(ValueSet.newBuilder().addAllItems(productItems).build())
+        .setVp(ValueSet.newBuilder().addAllItems(prophylaxisItems).build())
+        .build();
+  }
+
+  private List<ValueSetItem> toValueSetItems(Map<String, ValueSetObject> valueSetValues) {
+    return valueSetValues.entrySet().stream().map(
+        entry -> (ValueSetItem.newBuilder()
+            .setKey(entry.getKey())
+            .setDisplayText(entry.getValue().getDisplay())).build())
+        .collect(Collectors.toList());
+  }
+
+  private <T> T readConfiguredJsonOrDefault(String path, String defaultPath, Class<T> rawType) {
+    if (!ObjectUtils.isEmpty(path)) {
+      try (InputStream jsonStream = resourceLoader.getResource(path).getInputStream()) {
+        return deserializeJsonToSimpleType(jsonStream, rawType);
+      } catch (IOException e) {
+        logger.error("Error reading {} from json {}.", rawType.getSimpleName(), path, e);
+      }
+    }
+    try (InputStream jsonStream = resourceLoader.getResource(defaultPath).getInputStream()) {
+      // fallback to default
+      return deserializeJsonToSimpleType(jsonStream, rawType);
+    } catch (IOException e) {
+      logger.error("We could not load the default {}. This shouldn't happen!", defaultPath, e);
+      throw new RuntimeException("We could not load the default " + defaultPath
+          + ". This shouldn't happen!", e);
```

Comment:
@roesslerj / @AliceBilc 
will this abort the distribution of keys or event-checkins?

Reply:
Will be addressed in future PR (already implemented)